### PR TITLE
Adding dav resource for avatars

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/26872

@rullzer @mario @marinofaggiana @AndyScherzinger @tobiasKaminsky Might be interesting for you. Then you don't need  private APIs anymore and can also fetch it via WebDAV ;)

### CODE REVIEW

**Comment by MariusBluem** [comment]
@schiessle for federated profile pictures in sharing dialog 😁

**Comment by rullzer** [comment]
So, few things.

* WHEE!
* The extension might be problematic for clients
* I'm not really happy this loads the image resource into memory again. We should just dump the file.

Anyway. Will look into it.

**Review by LukasReschke — CHANGES_REQUESTED** [review]
Will do some changes to satisfy coverage and CI style.

**Review by LukasReschke — APPROVED** [review]

File: apps/dav/lib/Avatars/AvatarNode.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	 * This is used to generate the url.
+	 *
+	 * @return string
+	 */
+	public function getName() {
+		return "$this->size.$this->ext";
+	}
+
+	public function get() {
+		$image = $this->avatar->get($this->size);
```
Comment (rullzer) [inline apps/dav/lib/Avatars/AvatarNode.php]: I don't like this as it gets an image. Instead of just a file...

**Review by rullzer — CHANGES_REQUESTED** [review]

**Comment by rullzer** [comment]
Also no caching headers are set. Which would mean that if we switch over this will result in a shitload of requests to the server again

**Comment by rullzer** [comment]
Altough I'm not sure how we can set caching headers....

File: apps/dav/lib/Avatars/AvatarNode.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		$res = $image->resource();
+
+		ob_start();
+		if ($this->ext === 'png') {
+			imagepng($res);
+		} else {
+			impagejpeg($res);
```
Comment (LukasReschke) [inline apps/dav/lib/Avatars/AvatarNode.php]: Message: Call to undefined function OCA\DAV\Avatars\impagejpeg()

@rullzer 😉 – Also time to add tests to this 😝

**Review by LukasReschke — CHANGES_REQUESTED** [review]

**Comment by LukasReschke** [comment]
> Altough I'm not sure how we can set caching headers....

Probably something like `\OCA\DAV\CardDAV\ImageExportPlugin`?

**Review by rullzer — APPROVED** [review]

**Comment by rullzer** [comment]
Ok lets get this in and fixup the remaining stuff later.

**Comment by MorrisJobke** [comment]
Tested and works 👍

**Pull request merged**